mingw h265_find_startcode warning

```c
src/h265/nal.c: In function ‘h265_find_startcode’:
src/h265/nal.c:129:30: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  129 |  const uint8_t *a = p + 4 - ((long)p & 3);
      |                              ^
cc1: all warnings being treated as errors
```
@alfredh can you have a look?

--- a/.github/workflows/mingw.yml
+++ b/.github/workflows/mingw.yml
@@ -1,0 +1,79 @@
+name: MinGW-w64 Test
+
+on: [push, pull_request]
+
+jobs:
+  MinGW-w64-build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: "install packages"
+      run: |
+        sudo apt-get update && sudo apt-get install -y mingw-w64
+    
+    - uses: actions/checkout@v2 # needed for pr checkout
+
+    - uses: sreimers/pr-dependency-action@v0.3
+      with:
+        name: re
+        repo: https://github.com/baresip/re
+        secret: ${{ secrets.GITHUB_TOKEN }}
+    
+    - uses: sreimers/pr-dependency-action@v0.3
+      with:
+        name: retest
+        repo: https://github.com/baresip/retest
+        secret: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: "clone baresip-win32 repo"
+      run: |
+        git clone https://github.com/baresip/baresip-win32.git
+        mv re baresip-win32/
+        mv retest baresip-win32/
+    
+    - uses: actions/checkout@v2
+      with:
+        path: baresip-win32/rem
+
+    - uses: actions/cache@v2
+      id: openssl
+      with:
+        path: baresip-win32/openssl
+        key: ${{ runner.os }}-mingw-openssl-1.1.1m
+
+    - name: "build openssl"
+      if: steps.openssl.outputs.cache-hit != 'true'
+      run: |
+        wget https://www.openssl.org/source/openssl-1.1.1m.tar.gz
+        tar -xzf openssl-1.1.1m.tar.gz
+        mv openssl-1.1.1m baresip-win32/openssl
+        make -j$(nproc) -C baresip-win32 openssl
+    
+    - name: "build"
+      run: make -j$(nproc) -C baresip-win32 retest
+    
+    - uses: actions/upload-artifact@v2
+      with:
+        name: retest-exe
+        path: baresip-win32/retest/retest.exe
+        retention-days: 1
+
+  wintest:
+    runs-on: windows-2019
+    needs: MinGW-w64-build
+
+    steps:
+    - uses: actions/download-artifact@v2
+    - uses: actions/checkout@v2
+      with:
+        path: baresip
+    
+    - uses: sreimers/pr-dependency-action@v0.3
+      with:
+        name: retest
+        repo: https://github.com/baresip/retest
+        secret: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: "run retest.exe"
+      run: mv retest-exe/retest.exe retest/ && cd retest && ./retest.exe
+      shell: bash

--- a/src/h265/nal.c
+++ b/src/h265/nal.c
@@ -126,7 +126,7 @@ bool h265_is_keyframe(enum h265_naltype type)
 
 static const uint8_t *h265_find_startcode(const uint8_t *p, const uint8_t *end)
 {
-	const uint8_t *a = p + 4 - ((long)p & 3);
+	const uint8_t *a = p + 4 - ((size_t)p & 3);
 
 	for (end -= 3; p < a && p < end; p++ ) {
 		if (p[0] == 0 && p[1] == 0 && p[2] == 1)

